lib/cylc/url.py

```diff
@@ -0,0 +1,54 @@
+# This code is derived from the cPython "urllib" module and retains its
+# original license.
+#
+# Python software and documentation are licensed under the
+# Python Software Foundation License Version 2.
```

Comment:
@dpmatthews, can you ok this.

Sadly I couldn't achieve the required changes via hot patching.

The original code can be found here (2.7) https://github.com/python/cpython/blob/8d21aa21f2cbc6d50aab3f420bb23be1d081dac4/Lib/urllib.py#L1269

And the license here: https://github.com/python/cpython/blob/v2.7.18/LICENSE#L62